packages/application/src/shell.ts

```diff
@@ -488,7 +488,14 @@ class ApplicationShell extends Widget {
     // Make a copy of all the widget in the dock panel (using `toArray()`)
     // before removing them because removing them while iterating through them
     // modifies the underlying data of the iterator.
-    each(toArray(this._dockPanel.widgets()), widget => { widget.close(); });
+    this.closeWidgets(toArray(this._dockPanel.widgets()));
+  }
+
+  /**
+   * Close widgets in the given array.
+   */
+  closeWidgets(widgets: Array<Widget>): void {
```

Comment:
I originally had all the code, even the selection of which widgets to close, in app shell.  I can move the closeWidgets function to docmanager-extension.

Should I also move closeAll() then?  If not, I could update closeAll() to use forEach instead of each.